src/components/TeacherComponents/hooks/downloadAndShareHooks/useResourceFormState.tsx

```diff
@@ -53,42 +58,31 @@ export const useResourceFormState = (props: UseResourceFormStateProps) => {
   const [selectAllChecked, setSelectAllChecked] = useState(false);
   const [isLocalStorageLoading, setIsLocalStorageLoading] = useState(true);
   const [schoolUrn, setSchoolUrn] = useState(0);
+  const { useUser } = useFeatureFlaggedClerk();
+  const authFlagEnabled = useFeatureFlagEnabled("use-auth-owa");
+  const { isSignedIn, user } = useUser();
 
-  const resources = (() => {
-    switch (props.type) {
-      case "share":
-        return props.shareResources;
-      case "download":
-        return props.downloadResources;
-      case "curriculum":
-        return props.curriculumResources;
-    }
-  })();
+  const [hasOnboardingDownloadDetails, setHasOnboardingDownloadDetails] =
+    useState(false);
 
-  const getInitialResourcesState = useCallback(() => {
-    if (props.type === "share") {
-      return (resources as LessonShareData["shareableResources"])
-        .filter((resource) => resource.exists)
-        .map((resource) => resource.type);
-    } else if (props.type === "download") {
-      return (resources as LessonDownloadsPageData["downloads"])
-        .filter((resource) => resource.exists && !resource.forbidden)
-        .map((resource) => resource.type);
-    } else if (props.type === "curriculum") {
-      return (resources as CurriculumDownload[]).map(
-        (resource) => resource.url,
+  useEffect(() => {
+    if (user != null) {
+      // as user has signed in with full onboarding journey on OWA
+      const newHasOnboardingDownloadDetails = Boolean(
+        authFlagEnabled && isSignedIn && user.publicMetadata?.owa?.isTeacher,
```

Comment:
`isTeacher` will only be true if they onboarded and selected "Yes" to the first question. Anyone that has signed up will have agreed to terms. 

Is there much point in displaying the school picker and terms if the user has already indicated that they're not a teacher and have agreed to terms already?

I reckon this should be `user.publicMetadata?.owa?.isOnboarded`

Reply:
The downloads page is set up so that a school is required to download, so if we hide the school picker for a  and the user submits the download there will be validation errors but no form to address them

Reply:
🤔 sounds like the download page might need to change to allow that. I'd imagine it would make maintaining the existing page alongside the feature flagged logic a nightmare. Maybe another ticket since this needs discussion?

Reply:
Yeah i think another ticket, i'll ask in the squad channel shortlt